src/sql/workbench/browser/modelComponents/radioCardGroup.component.ts

```diff
@@ -141,7 +141,10 @@ export default class RadioCardGroup extends ComponentBase implements IComponent,
 		this._changeRef.detectChanges();
 		this.fireEvent({
 			eventType: ComponentEventType.onDidChange,
-			args: cardId
+			args: {
+				cardId,
+				card: Object.assign({}, this.getCardById(cardId))
```

Comment:
This isn't really doing a clone - for example the descriptions array is still going to reference the same array (same with any objects)

Not a huge deal currently since this goes over the RPC layer anyways so isn't connected regardless - but you should use https://github.com/microsoft/azuredatastudio/blob/main/src/vs/base/common/objects.ts#L8 regardless for anyone that might hook into the event on the main thread side.

Reply:
Good catch, thought all the elements under it are just strings and stuff.

Reply:
https://github.com/microsoft/azuredatastudio/blob/main/src/sql/azdata.proposed.d.ts#L151

descriptions is an array and the icon can be an object so that's not really true 😄